File: VB/CustomEditor/MyCheckedComboBoxEdit.vb
Imports System
Imports System.ComponentModel
Imports System.Drawing
Imports System.Windows.Forms
Imports DevExpress.XtraEditors
Imports DevExpress.XtraEditors.Repository
Imports DevExpress.XtraEditors.Registrator

Namespace WindowsApplication1
	''' <summary>
	''' MyCheckedComboBoxEdit is a descendant from CheckedComboBoxEdit.
	''' It displays a dialog form below the text box when the edit button is clicked.
	''' </summary>
	Public Class MyCheckedComboBoxEdit
		Inherits CheckedComboBoxEdit

		Shared Sub New()
			RepositoryItemMyCheckedComboBoxEdit.Register()
		End Sub



		Public Overrides ReadOnly Property EditorTypeName() As String
			Get
				Return RepositoryItemMyCheckedComboBoxEdit.EditorName
			End Get
		End Property
		<DesignerSerializationVisibility(DesignerSerializationVisibility.Content)>
		Public Shadows ReadOnly Property Properties() As RepositoryItemMyCheckedComboBoxEdit
			Get
				Return TryCast(MyBase.Properties, RepositoryItemMyCheckedComboBoxEdit)
			End Get
		End Property

		Public Overrides Property EditValue() As Object
			Get
'INSTANT VB NOTE: The local variable editValue was renamed since Visual Basic will not allow local variables with the same name as their enclosing function or property:
				Dim editValue_Conflict As Object = MyBase.EditValue
				If isParsing Then
					editValue_Conflict = Properties.MyFormatEditValue(editValue_Conflict)
				End If
				Return editValue_Conflict
			End Get
			Set(ByVal value As Object)
				MyBase.EditValue = value
			End Set
		End Property

		Private isParsing As Boolean

		Protected Overrides Sub DoSynchronizeEditValueWithCheckedItems()
			isParsing = True
			MyBase.DoSynchronizeEditValueWithCheckedItems()
			isParsing = False
		End Sub

	End Class
End Namespace

File: CS/CustomEditor/MyCheckedComboBoxEdit.cs
using System;
using System.ComponentModel;
using System.Drawing;
using System.Windows.Forms;
using DevExpress.XtraEditors;
using DevExpress.XtraEditors.Repository;
using DevExpress.XtraEditors.Registrator;

namespace WindowsApplication1 {
	/// <summary>
	/// MyCheckedComboBoxEdit is a descendant from CheckedComboBoxEdit.
	/// It displays a dialog form below the text box when the edit button is clicked.
	/// </summary>
	public class MyCheckedComboBoxEdit : CheckedComboBoxEdit {
		static MyCheckedComboBoxEdit() {
			RepositoryItemMyCheckedComboBoxEdit.Register();
		}



		public override string EditorTypeName { 
			get { return RepositoryItemMyCheckedComboBoxEdit.EditorName; } 
		}
		[DesignerSerializationVisibility(DesignerSerializationVisibility.Content)]
		public new RepositoryItemMyCheckedComboBoxEdit Properties { 
			get { return base.Properties as RepositoryItemMyCheckedComboBoxEdit; } 
		}

        public override object EditValue
        {
            get
            {
                object editValue = base.EditValue;
                if (isParsing)
                    editValue = Properties.MyFormatEditValue(editValue);
                return editValue;
            }
            set
            {
                base.EditValue = value;
            }
        }

        bool isParsing;

        protected override void DoSynchronizeEditValueWithCheckedItems()
        {
            isParsing = true;
            base.DoSynchronizeEditValueWithCheckedItems();
            isParsing = false;
        }

	}
}
